Turn off the filtering of hop nodes from the Instance endpoints

jbradberry opened the pull request:
<!--- changelog-entry
# Fill in 'msg' below to have an entry automatically added to the next release changelog.
# Leaving 'msg' blank will not generate a changelog entry for this PR.
# Please ensure this is a simple (and readable) one-line string.
---
msg: ""
-->

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
This change re-enables hop nodes being accessible in all API endpoints except for the health check.  This enables having a UI detail page that users can reach from the mesh visualizer.

As a side-effect, this deals with the problem in #11575.

related #11588 
related #11575 

<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request
 - Bugfix Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - API

jbradberry committed:
Turn off the filtering of hop nodes from the Instance endpoints

except for the health check.

AlanCoding reviewed (approved):
I still see at least 2 things we need to do:
1. in the InstanceSerializer, avoid linking to the health check if it's a hop node
2. in the cluster_node_heartbeat, stop filtering out the hop nodes

doing (2) might involve adding some more exceptions, because the idea is that we don't run health checks, but we can still mark it as error if it hasn't updated its `last_seen` timestamp in ages.

Anyway, these are some good stability-oriented bug fixes and I say :+1:

jbradberry committed:
Handle receptorctl advertisements for hop nodes

counting it towards their heartbeat.  Also, leave off the link to the
health check endpoint from hop node Instances.

jbradberry committed:
Adjust the list_instances command to show hop nodes

with appropriate attributes removed or added.

jbradberry committed:
Adjust register_queue command to not allow hop nodes to be added

AlanCoding commented on awx/main/tasks/system.py:
```diff
@@ -454,6 +451,10 @@ def inspect_execution_nodes(instance_list):
 
+            # Make sure that hop nodes don't fall through and have the execution_node_health_check task applied
+            if not any(cmd['WorkType'] == 'ansible-runner' for cmd in ad['WorkCommands'] or []):
+                continue
```
yeah, this is probably not needed anymore, but it looks good as a defensive coding measure.

jbradberry replied:
Marking this for posterity: we decided on a clearer, more-readable pattern for these checks.

AlanCoding commented on awx/main/tasks/system.py:
```diff
@@ -482,7 +483,6 @@ def cluster_node_heartbeat():
     for inst in instance_list:
         if inst.hostname == settings.CLUSTER_HOST_ID:
             this_inst = inst
-            instance_list.remove(inst)
```
Removing this, I would like to avoid `inst` getting in the `lost_instances` list. By definition, if you are running this code, you are not a lost instance. Any issues will be handled in `this_isnt.local_health_check()` which is a separate block of code. I'm a little worried that the order of operations might jerk its error status back and fourth.

jbradberry replied:
Dealt with, for now.  A real cleanup of this code seems in order, though.

AlanCoding reviewed (approved):
Left one minor logic change request, otherwise everything looks good to go :+1:

jbradberry committed:
Make sure to filter out control-plane nodes in inspect_execution_nodes

Also, make sure that the cluster host doesn't get marked as lost by
this machinery.
